python/mlcroissant/mlcroissant/_src/operation_graph/execute.py

```diff
@@ -124,3 +132,37 @@ def read_all_files():
                 "An error occured during the streaming generation of the dataset, more"
                 f" specifically during the operation {operation}"
             ) from e
+
+
+def execute_operations_in_beam(
+    pipeline: beam.Pipeline, record_set: str, operations: Operations
+):
+    """Executes operations in Beam.
+
+    This is only possible for streamable datasets, i.e. datasets that result from
```

Comment:
NIT: "This is only possible for streamable datasets" --> Not sure if this is clear from the code... the check for non-streamable datasets (and the consequent error) happens in beam_reader no? So what would happen here if a non-streamable dataset would be given as input here? Where would the function fail?